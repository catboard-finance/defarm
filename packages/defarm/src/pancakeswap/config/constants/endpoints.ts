export const GRAPH_API_PROFILE = process.env.REACT_APP_GRAPH_API_PROFILE || "https://api.thegraph.com/subgraphs/name/pancakeswap/profile"
export const GRAPH_API_PREDICTION = process.env.REACT_APP_GRAPH_API_PREDICTION || "https://api.thegraph.com/subgraphs/name/pancakeswap/prediction"
export const ARCHIVED_NODE = process.env.REACT_APP_ARCHIVED_NODE || "https://little-long-pine.bsc.quiknode.pro/"
