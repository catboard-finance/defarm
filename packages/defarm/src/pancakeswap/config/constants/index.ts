export { default as farmsConfig } from './farms'
export { default as poolsConfig } from './pools'
export { default as tokensConfig } from './tokens'
