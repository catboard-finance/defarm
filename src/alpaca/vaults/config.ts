interface IInput {
  VAULT_SYMBOL: string;
  ADD_STRATEGIES: string[];
  IS_ENABLE: boolean;
}

export const TARGETED_VAULT_CONFIG: Array<IInput> = [
  {
    VAULT_SYMBOL: "ibWBNB",
    ADD_STRATEGIES: [
      "0x4c7a420142ec69c7Df5c6C673D862b9E030743bf",
      "0x5cB454fc86068e710212FBECBC93070b90011F2B",
      "0xeBb8BA21A3703ab30187D3EEC02A3Bc62894970D",
      "0xB9B8766B65636779C3B169B9a18e0A708F91c610",
      "0xB0951EB5eCd9948AAA8Eb76D1061361F592BA029",
      "0xA7559bB0235a1c6003D0E48d2cFa89a6C8748439",
    ],
    IS_ENABLE: true,
  },
  {
    VAULT_SYMBOL: "ibBUSD",
    ADD_STRATEGIES: [
      "0x4c7a420142ec69c7Df5c6C673D862b9E030743bf",
      "0x5cB454fc86068e710212FBECBC93070b90011F2B",
      "0xeBb8BA21A3703ab30187D3EEC02A3Bc62894970D",
      "0x3fC149995021f1d7AEc54D015Dad3c7Abc952bf0",
      "0x38912684b1d20Fe9D725e8B39c39458Fac5A4833",
      "0x61e58dE669d842C2d77288Df629af031b3283c81",
    ],
    IS_ENABLE: true,
  },
  {
    VAULT_SYMBOL: "ibETH",
    ADD_STRATEGIES: [
      "0x4c7a420142ec69c7Df5c6C673D862b9E030743bf",
      "0x5cB454fc86068e710212FBECBC93070b90011F2B",
      "0xeBb8BA21A3703ab30187D3EEC02A3Bc62894970D",
      "0xCB459b4504d10445760095C59c394EA45715d7a5",
      "0x86547E01b7F1BAc1F4cE80A4964829009D2dE1Cc",
      "0xD58b9626d941cA2d31b55a43045d34A87B32cEd3",
    ],
    IS_ENABLE: true,
  },
  {
    VAULT_SYMBOL: "ibALPACA",
    ADD_STRATEGIES: [
      "0x4c7a420142ec69c7Df5c6C673D862b9E030743bf",
      "0x5cB454fc86068e710212FBECBC93070b90011F2B",
      "0xeBb8BA21A3703ab30187D3EEC02A3Bc62894970D",
      "0xa964FCd9a434CB4C68bFE25E77D1F2Cd5D9679a8",
      "0x09176545F3c013142b69477D7De2E7F4BAa2bB3a",
      "0x462BC565b5486E76503DaB62D96937842F207AAa",
    ],
    IS_ENABLE: true,
  },
  {
    VAULT_SYMBOL: "ibUSDT",
    ADD_STRATEGIES: [
      "0x4c7a420142ec69c7Df5c6C673D862b9E030743bf",
      "0x5cB454fc86068e710212FBECBC93070b90011F2B",
      "0xeBb8BA21A3703ab30187D3EEC02A3Bc62894970D",
      "0x5f94f61095731b669b30ed1f3f4586BBb51f4001",
      "0x50380Ac8DA73D73719785F0A4433192F4e0E6c90",
      "0xcE37fD1Ff0A6cb4A6A59cd46CCf55D5Dc70ec585",
    ],
    IS_ENABLE: true,
  },
  {
    VAULT_SYMBOL: "ibBTCB",
    ADD_STRATEGIES: [
      "0x4c7a420142ec69c7Df5c6C673D862b9E030743bf",
      "0x5cB454fc86068e710212FBECBC93070b90011F2B",
      "0xeBb8BA21A3703ab30187D3EEC02A3Bc62894970D",
      "0xe862D45abdF7ea38F75dd0c7164B19FaEd057130",
      "0x30A937B9d22d71e58Ad9dC96a6A3d552B9C0724e",
      "0xaC712F4Fc61ab96Aa9A1AdF3977b808789aA6682",
    ],
    IS_ENABLE: true,
  },
  {
    VAULT_SYMBOL: "ibTUSD",
    ADD_STRATEGIES: [
      "0x4c7a420142ec69c7Df5c6C673D862b9E030743bf",
      "0x5cB454fc86068e710212FBECBC93070b90011F2B",
      "0xeBb8BA21A3703ab30187D3EEC02A3Bc62894970D",
      "0xA8F37daF3D290F636f0B79E47ea50aB7f7A82d51",
      "0x67a2CdB9F0760663B5E70c1517f1a603BA3F50f0",
      "0x8cE75fF793D7832302BeA91c275e2509060DfEaA",
    ],
    IS_ENABLE: true,
  },
];

export const ALL_STRATEGY_ADDRESSES = TARGETED_VAULT_CONFIG.map(e => e.ADD_STRATEGIES).flat()
export const ALL_STRATEGY_LOWER_ADDRESSES = ALL_STRATEGY_ADDRESSES.map(e => e.toLowerCase());
