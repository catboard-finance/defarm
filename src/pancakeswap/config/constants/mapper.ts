import { farmsConfig } from './index'
import { FarmConfig } from './types'

interface FarmSymbolMap {
  [lpSymbol: string]: FarmConfig;
}

const _farmsSymbolMap = {}
farmsConfig.forEach(farmConfig => {
  _farmsSymbolMap[farmConfig.lpSymbol] = _farmsSymbolMap[farmConfig.lpSymbol] || farmConfig
})

export const farmsSymbolMap: FarmSymbolMap = _farmsSymbolMap

interface FarmAddressMap {
  [address: string]: FarmConfig;
}

const _farmsAddressMap = {}
farmsConfig.forEach(farmConfig => {
  _farmsAddressMap[farmConfig.lpAddresses[56]] = _farmsAddressMap[farmConfig.lpAddresses[56]] || farmConfig
})

export const farmsAddressMap: FarmAddressMap = _farmsAddressMap
