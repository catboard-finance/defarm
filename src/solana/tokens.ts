interface ISolanaToken {
  symbol: string
  name: string
  address: string
  decimals: string
}

export const getSolanaSymbolInfoByAddress = (address: string): ISolanaToken => {
  const whitelist = require('./whitelist')
  const token = whitelist.find((e) => e.address === address)
  return token
}
