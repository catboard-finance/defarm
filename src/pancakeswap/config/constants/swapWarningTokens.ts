import tokens from './tokens'
import { Address } from './types'

const { bondly, safemoon } = tokens

interface WarningToken {
  symbol: string
  address: Address
}

interface WarningTokenList {
  [key: string]: WarningToken
}

const SwapWarningTokens = <WarningTokenList>{
  safemoon,
  bondly,
}

export default SwapWarningTokens
