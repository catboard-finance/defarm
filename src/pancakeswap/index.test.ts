import { fetchFarmsWithAPRBySymbols, fetchFarmUserDataAsync, fetchTokenUSDPricesBySymbols, getSupportedUSDSymbols } from ".";
import { farmsConfig } from "./config/constants";
import { farmsSymbolMap } from "./config/constants/mapper";
import tokens from "./config/constants/tokens";

describe('🍰 global data', () => {
  beforeAll(() => {
    jest.spyOn(console, 'warn').mockImplementation(() => { });
    jest.spyOn(console, 'log').mockImplementation(() => { });
  });

  it('has supported symbol list', async () => {
    const supportedSymbols = getSupportedUSDSymbols()
    expect(supportedSymbols).toMatchSnapshot()
  });

  it('can get CAKE price', async () => {
    const [cake] = await fetchTokenUSDPricesBySymbols(['CAKE'])

    expect(cake.address).toMatchSnapshot()
    expect(parseFloat(cake.busdPrice)).toBeGreaterThan(1)
  }, 10000);

  it('can get ALPACA, ETH, BNB prices', async () => {
    const [alpaca, eth, bnb] = await fetchTokenUSDPricesBySymbols(['ALPACA', 'ETH', 'BNB'])
    // console.log(alpaca, eth, bnb)

    expect(alpaca.symbol).toEqual(tokens.alpaca.symbol)
    expect(alpaca.address).toEqual(tokens.alpaca.address[56])
    expect(parseFloat(alpaca.busdPrice)).toBeGreaterThan(0)

    expect(eth.symbol).toEqual(tokens.eth.symbol)
    expect(eth.address).toEqual(tokens.eth.address[56])
    expect(parseFloat(eth.busdPrice)).toBeGreaterThan(0)

    expect(bnb.symbol).toEqual(tokens.bnb.symbol)
    expect(bnb.address).toEqual(tokens.wbnb.address[56])
    expect(parseFloat(bnb.busdPrice)).toBeGreaterThan(0)
  }, 10000);

  it('can get CAKE-BNB LP info', async () => {
    const [cake_bnb] = await fetchFarmsWithAPRBySymbols(['CAKE-BNB LP'])

    expect(cake_bnb.lpSymbol).toEqual('CAKE-BNB LP')
    expect(cake_bnb.lpAddresses[56]).toEqual(farmsSymbolMap['CAKE-BNB LP'].lpAddresses[56])
    expect(parseFloat(cake_bnb.cakeRewardsApr)).toBeGreaterThan(0)
    expect(parseFloat(cake_bnb.lpRewardsApr)).toBeGreaterThan(0)
    expect(parseFloat(cake_bnb.mintRate)).toBeGreaterThan(0)
  }, 10000);

  it('can return null for unknown price', async () => {
    const [alpaca, not_exist] = await fetchTokenUSDPricesBySymbols(['ALPACA', 'NOT_EXIST'])

    expect(not_exist.symbol).toEqual('NOT_EXIST')
    expect(not_exist.address).toBeNull()
    expect(not_exist.busdPrice).toBeNull()

    expect(alpaca.symbol).toEqual(tokens.alpaca.symbol)
    expect(alpaca.address).toEqual(tokens.alpaca.address[56])
    expect(parseFloat(alpaca.busdPrice)).toBeGreaterThan(0)
  }, 10000);
})

describe('🍰 user data', () => {
  it('can get user data', async () => {
    const account = "0x8155430e4860e791aeddb43e4764d15de7e0def1"
    const pid = farmsConfig[0].pid
    const userData = await fetchFarmUserDataAsync(farmsConfig, { account, pids: [pid] })

    expect(parseFloat(userData[0].tokenBalance)).toBeGreaterThan(0)
  }, 10000)
});
