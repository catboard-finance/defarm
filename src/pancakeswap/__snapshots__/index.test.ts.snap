// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`üç∞ global data can get CAKE price 1`] = `"0x0E09FaBB73Bd3Ade0a17ECC321fD13a19e81cE82"`;

exports[`üç∞ global data has supported symbol list 1`] = `
Array [
  "ETH",
  "RAMP",
  "POTS",
  "TUSD",
  "BTT",
  "TRX",
  "WIN",
  "IF",
  "PMON",
  "TITAN",
  "ADX",
  "FORM",
  "ORBS",
  "O3",
  "AMPL",
  "MX",
  "bCFX",
  "QKC",
  "MTRG",
  "WELL",
  "DERI",
  "CHR",
  "CAKE",
  "WMASS",
  "UBXT",
  "BTR",
  "ONE",
  "OIN",
  "KUN",
  "MATH",
  "xMARK",
  "BTCB",
  "DFD",
  "ALPACA",
  "HAKKA",
  "HOO",
  "TXL",
  "FOR",
  "NULS",
  "DEXE",
  "TPT",
  "IOTX",
  "LINA",
  "UST",
  "USDC",
  "DAI",
  "VAI",
  "USDT",
  "HAKKA",
  "HOO",
  "TXL",
  "FOR",
  "NULS",
  "RAMP",
  "DEXE",
  "TPT",
  "xMARK",
  "IOTX",
  "LINA",
  "UST",
  "USDC",
  "DAI",
  "VAI",
  "USDT",
  "HAKKA",
  "HOO",
  "TXL",
  "FOR",
  "NULS",
  "RAMP",
  "DEXE",
  "TPT",
  "xMARK",
  "IOTX",
  "LINA",
  "UST",
  "USDC",
  "DAI",
  "VAI",
  "USDT",
]
`;
